File: spec/test_optional_rule_component.c
#include "testparser.h"
#include <assert.h>
#include <string.h>

int main()
{
    char const * input = "b";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    input = "abcd";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    input = "abdc";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    return 0;
}

File: spec/test_optional_rule_component.rs
use testparser::*;

fn main() {
    for input in [&b"b"[..], &b"abcd"[..], &b"abdc"[..]] {
        let mut context = p_context_new(input);
        assert_eq!(P_SUCCESS, p_parse(&mut context));
        p_context_delete(context);
    }
}
